setup.ts

```diff
@@ -0,0 +1,379 @@
+const dotenv = require("dotenv");
+const fs = require("fs");
+const cryptolib = require("crypto");
+const inquirer = require("inquirer");
+const mongodb = require("mongodb");
+const { exec } = require("child_process");
+
+dotenv.config();
+
+// Check if all the fields in .env.sample are present in .env
+function checkEnvFile(): void {
+  const env = dotenv.parse(fs.readFileSync(".env"));
+  const envSample = dotenv.parse(fs.readFileSync(".env.sample"));
+  const misplaced = Object.keys(envSample).filter((key) => !(key in env));
+  if (misplaced.length > 0) {
+    console.log("Please copy the contents of .env.sample to .env file");
+    abort();
+  }
+}
+
+// Generate and update the access and refresh token secrets in .env
+async function accessAndRefreshTokens(
+  accessTokenSecret: string | null,
+  refreshTokenSecret: string | null
+): Promise<void> {
+  const config = dotenv.parse(fs.readFileSync(".env"));
+
+  if (accessTokenSecret == null) {
+    accessTokenSecret = cryptolib.randomBytes(32).toString("hex");
+    config.ACCESS_TOKEN_SECRET = accessTokenSecret;
+    fs.writeFileSync(".env", "");
+    for (const key in config) {
+      fs.appendFileSync(".env", `${key}=${config[key]}\n`);
+    }
+  }
+
+  if (refreshTokenSecret == null) {
+    refreshTokenSecret = cryptolib.randomBytes(32).toString("hex");
+    config.REFRESH_TOKEN_SECRET = refreshTokenSecret;
+    fs.writeFileSync(".env", "");
+    for (const key in config) {
+      fs.appendFileSync(".env", `${key}=${config[key]}\n`);
+    }
+  }
+}
+
+// Check the connection to MongoDB with the specified URL.
+async function checkConnection(url: string): Promise<boolean> {
+  let response = false;
+  const client = new mongodb.MongoClient(url, {
+    useNewUrlParser: true,
+    useUnifiedTopology: true,
+    serverSelectionTimeoutMS: 1000,
+  });
+
+  console.log("\nChecking MongoDB connection....");
+
+  try {
+    await client.connect();
+    console.log("\nConnection to MongoDB successful! 🎉");
+    response = true;
+  } catch (error) {
+    console.log(`\nConnection to MongoDB failed. Please try again.\n`);
+  }
+  client.close();
+  return response;
+}
+
+//Mongodb url prompt
+async function askForMongoDBUrl(): Promise<string> {
+  const { url } = await inquirer.prompt([
+    {
+      type: "input",
+      name: "url",
+      message: "Enter your MongoDB URL:",
+    },
+  ]);
+
+  return url;
+}
+
+// Get the mongodb url
+async function mongoDB(): Promise<void> {
+  let DB_URL = process.env.MONGO_DB_URL;
+
+  try {
+    let isConnected = false,
+      url = "";
+    while (!isConnected) {
+      url = await askForMongoDBUrl();
+      isConnected = await checkConnection(url);
+    }
+
+    DB_URL = url;
+    const config = dotenv.parse(fs.readFileSync(".env"));
+    config.MONGO_DB_URL = DB_URL;
+    process.env.MONGO_DB_URL = DB_URL;
```

Comment:
is this line really needed?

Reply:
yes, otherwise I will have to read MongoDB URL again during data import

Reply:
@saiteja13427 I didn't get that. Why would u alter the environment variable? You're using `process.env.MONGO_DB_URL` in data import function, so what do you mean by reading it again?

Reply:
@xoldyckk Because I am setting it here, I am able to read it in data import function. If I don't set this env variable here, I will have to read the URL again from .env file in data import function.
So, to avoid the read, I am setting the URL to this env variable and using it to import data.

Reply:
@saiteja13427 U ask for mongodb url from user, save it to env file. Are u saying the changes in env file not reflect in `process.env` global variable unless the whole process is restarted?

Reply:
Yes exactly!

Reply:
Hmmm, this is a really hacky way of doing it but whatever. Add comments at proper places to let the people know why you're mutating it.

Reply:
@xoldyckk Alright let me push a comment there!

Reply:
@xoldyckk Added a comment there!